swarm-derive/src/lib.rs

```diff
@@ -171,35 +179,44 @@ fn build_struct(ast: &DeriveInput, data_struct: &DataStruct) -> TokenStream {
             })
     };
 
-    // Build the list of statements to put in the body of `inject_connection_established()`.
-    let inject_connection_established_stmts = {
-        data_struct.fields.iter().enumerate().map(move |(field_n, field)| {
-            match field.ident {
-                Some(ref i) => quote!{ self.#i.inject_connection_established(peer_id, connection_id, endpoint, errors, other_established); },
-                None => quote!{ self.#field_n.inject_connection_established(peer_id, connection_id, endpoint, errors, other_established); },
-            }
-        })
+    // Build the list of statements to put in the body of `on_swarm_event()`
+    // for the `FromSwarm::ConnectionEstablished` variant.
+    let on_connection_established_stmts = {
+        data_struct
+            .fields
+            .iter()
+            .enumerate()
+            .map(|(field_n, field)| match field.ident {
+                Some(ref i) => quote! { self.#i.on_swarm_event(
```

Comment:
`swarm-derive` will need to delegate to the `inject_*` methods of the corresponding `NetworkBehaviour` implementations, right? Otherwise our non-breaking deprecation doesn't work.

Reply:
yeah right, thanks for noticing Max, I think I addressed all the cases but ptal.